### PR TITLE
Improved StopTask in case that container was lost outside nomad

### PR DESCRIPTION
**Pull request by towe75**
Handle stopped and unknown/lost containers correctly while stopping a task.
The previous version let nomad run circles around lost containers. This happend when a group was reallocated to another node when a host was rebooted. After rebook, nomad tried to stop the already stopped container etc.  
I found that the docker driver has a similar if/else chain in StopTask, so i reassembled it here.

### CODE REVIEW

**Review by drewbailey — APPROVED** [review]
looks good to me!

**Comment by towe75** [comment]
@drewbailey : please check again. I changed the code because of some comments in #116 . Now, we're simply using the existing "ignore" option in the podman container-stop API call.

File: api/container_stop.go [hunk 1]
```diff
@@ -10,15 +10,19 @@ import (
 // timeout value.  The timeout value the time before a forcible stop to the container is applied.
 // If the container cannot be found, a [ContainerNotFound](#ContainerNotFound)
 // error will be returned instead.
-func (c *API) ContainerStop(ctx context.Context, name string, timeout int) error {
+func (c *API) ContainerStop(ctx context.Context, name string, timeout int, ignoreStopped bool) error {
```
Comment (drewbailey) [inline api/container_stop.go:13]: Not necessary for now, but this seems like something we may eventually want to support a generic QueryOptions struct for as the list of options grows

**Review by drewbailey — APPROVED** [review]

File: api/container_stop.go (hunk 1 above)
Comment (towe75) [reply api/container_stop.go:13]: Yep, let's see if it grows again.

**Pull request merged**